.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -190,26 +191,27 @@ jobs:
       run: rustup component add clippy rustfmt
 
     - name: "Rust tools cache"
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       id: rust-tools-cache
       with:
         path: |
           ${{ matrix.cargo_dir }}/bin/sccache${{ matrix.exe_suffix }}
           ${{ matrix.cargo_dir }}/bin/mdbook${{ matrix.exe_suffix }}
           ${{ matrix.cargo_dir }}/bin/mdbook-linkcheck${{ matrix.exe_suffix }}
-        key: ${{ matrix.os }}_sccache-0.4.1_mdbook-0.4.28_mdbook-linkcheck-0.7.7
+        key: ${{ matrix.os }}_sccache-0.7.6_mdbook-0.4.36_mdbook-linkcheck-0.7.7
     - name: "Build Rust tools"
       if: steps.rust-tools-cache.outputs.cache-hit != 'true'
       run: cargo install --locked sccache mdbook mdbook-linkcheck
 
     - name: "Compiler cache"
-      uses: actions/cache@v3
+      # Always cache and save all branches
+      # otherwise if we only restore here and only save the main branch
+      # we don't update the cache until something reaches main which
+      # causes an eviction of the cache if the cache isn't accessed for 1 week
```

Comment:
should we schedule a "nightly" run of the CI every few days, this would ensure that things don't break from newer versions of packages (as we have had before) and cause the cache not to expire.

Looks like it'd be adding an `on` block to this file https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

```
on:
  schedule:
    # * is a special character in YAML so you have to quote this string
    - cron:  '30 5,17 * * *'
```

Reply:
Sounds like a good idea.

But that can be a separate PR I think.